pkg/controller/statefulset/stateful_pod_control.go

```diff
@@ -112,23 +112,26 @@ func (spc *realStatefulPodControl) UpdateStatefulPod(set *apps.StatefulSet, pod
 				return err
 			}
 		}
-		// if the Pod is not dirty do nothing
+		// if the Pod is not dirty, do nothing
 		if consistent {
 			return nil
 		}
 
 		attemptedUpdate = true
 		// commit the update, retrying on conflicts
-		_, err := spc.client.Core().Pods(set.Namespace).Update(pod)
-		if err == nil {
-			return nil
+		_, updateErr := spc.client.Core().Pods(set.Namespace).Update(pod)
+		if updateErr == nil {
+			return updateErr
 		}
-		updateErr := err
 
 		if updated, err := spc.podLister.Pods(set.Namespace).Get(pod.Name); err == nil {
 			// make a copy so we don't mutate the shared cache
 			if copy, err := api.Scheme.DeepCopy(updated); err == nil {
-				pod = copy.(*v1.Pod)
```

Comment:
@kargakis I will give up these modifications. And as to "updateErr" and "adding comma", do you agree?